inference-engine/src/plugin_api/cpp_interfaces/impl/ie_infer_request_internal.hpp

```diff
@@ -228,13 +228,29 @@ class InferRequestInternal : virtual public IInferRequestInternal {
      * @param serial Whether to use multiple threads to execute the step
      */
     void execDataPreprocessing(InferenceEngine::BlobMap& inputs, bool serial = false) {
-        for (auto& input : inputs) {
-            // If there is a pre-process entry for an input then it must be pre-processed
-            // using preconfigured resize algorithm.
-            auto it = _preProcData.find(input.first);
-            if (it != _preProcData.end()) {
-                _preProcData[input.first]->execute(input.second, _networkInputs[input.first]->getPreProcess(), serial,
-                                                   m_curBatch);
+        for (auto& pp_result : inputs) {
+            auto const& input_name    = pp_result.first;
+            auto & result_blob        = pp_result.second;
+            auto const& ir_input_blob = _inputs[input_name];
```

Comment:
What happens if `result_blob` (`inputs[input_name]`) and `ir_input_blob`(`_inputs[input_name]`) point to the one blob?

Reply:
it's handled in the `device_specific_preproc_req` flag

Reply:
not relevant any more - patch is reworked